### PR TITLE
fix: check for duplicate configs and secrets within a module

### PR DESCRIPTION
**Pull request by deniseli**
Contributes to one part of https://github.com/TBD54566975/ftl/issues/1121

### CODE REVIEW

File: go-runtime/compile/schema.go [hunk 1]
```diff
@@ -257,6 +257,21 @@ func parseConfigDecl(pctx *parseContext, node *ast.CallExpr, fn *types.Func) {
 			Type: st,
 		}
 	}
+
+	// Check for duplicates
+	for _, d := range pctx.module.Decls {
+		c, ok := d.(*schema.Config)
+		if ok && c.Name == name && c.Type.String() == st.String() {
+			pctx.errors.add(errorf(node, "duplicate config declaration"))
```
Comment (alecthomas) [inline go-runtime/compile/schema.go:265]: Let's report the location of the other declaration in the message (here and below).

**Comment by alecthomas** [comment]
The original issue was just about preventing duplicates in the schema, not necessarily in the code. I think this makes sense though, as it's a bit of a code smell to redeclare something in multiple locations.

**Review by alecthomas — APPROVED** [review]

**Comment by alecthomas** [comment]
You'll need to roll this out to the PFI repo, which will be ... interesting.

**Comment by deniseli** [comment]
Notes to self (and anyone curious :) )

* Add check for duplicate databases
* Add checks to `backend` for all 3 types
* Do not merge until after Matt and Wes's huge changes go in because we will want to strategize on how to roll this out to PFI, and that'll be simpler once their changes are done (or at least discussed/planned/strategized)

**Comment by alecthomas** [comment]
Don't delay merges, just get them in.

**Comment by deniseli** [comment]
In that case, merging now! I'll move my TODOs to the issue to make sure I don't forget them.

File: go-runtime/compile/schema.go (hunk 1 above)
Comment (deniseli) [reply go-runtime/compile/schema.go]: great idea - done

**Comment by deniseli** [comment]
Got it, sounds good! I'll merge this to start and get the rest of it done in a following PR. I'll move my TODOs to the issue for better tracking.

**Pull request merged**